doc/modules/preprocessing.rst

```diff
@@ -411,3 +411,122 @@ values than observed values.
+the ``scale`` parameter described later, but there is no substitute for being
+careful.
+
+
+Custom label distribution
+-------------------------
+The three built-in sampling options above produce a balanced dataset, but
+unbalancing the data by passing a ``method=dict`` is also supported. For
+instance, in the previous example, passing ``method={0: .5, 1: .25, 2: .25}``
+gives a dataset where a 0 sample is twice as likely as a 1 or 2.
+
+Skew the distribution with a dict::
+
+  >>> import numpy as np
+  >>> from sklearn.preprocessing.resample import resample_labels
+  >>> y = np.concatenate([np.repeat(0,100), np.repeat(1,125), np.repeat(2,150)])
+  >>> indices = resample_labels(y, method={0: .5, 1: .25, 2: .25}, random_state=4)
+  >>> print(np.bincount(y[indices]))
+  [179 100  96]
+
+
+Resizing the dataset
+--------------------
+Changing the size of a dataset is also supported, which can lead to interesting
+training possibilities. Perhaps the full dataset is very large and will take a
+long time to train. Setting ``scaling=3000``, for example, will output 3000
+samples with the desired label distribution, while setting ``scaling`` to a
+float will scale the original number of samples, e.g. ``scaling=.5`` will
+output half the number of samples in the original dataset, while
+``scaling=3.0`` will triple the dataset.
+
+Scaling up the size of a training set is also useful for making sure that
+a machine learning algorithm can handle training and predicting large amounts
+of data in a reasonable amount of time.
+
+Take scaling up the iris dataset from 150 samples to 150,000 as an example::
+
+    >>> import numpy as np
+    >>> from sklearn import svm, datasets
+    >>> from sklearn.preprocessing.resample import resample_labels
+    >>> iris = datasets.load_iris()
+    >>> X = iris.data
+    >>> y = iris.target
+    >>> indices = resample_labels(y, scaling=1000.0)
+    >>> X0 = iris.data[indices]
+    >>> y0 = iris.target[indices]
+    >>> svc = svm.SVC(kernel='linear', C=1.0).fit(X, y)
+    >>> svc = svm.SVC(kernel='linear', C=1.0).fit(X0, y0)
+
+Training is almost instantaneous with 150 samples, but can take over ten
+seconds with 1000x the data.
+
+Other resampling options
+------------------------
+Sampling can be done with or without replacement, with the default being
+``replace=False`` for sampling without replacement. In order for the
+resizing feature to work with sampling without replacement, if we would
+run out of samples, instead the sample pool starts over with the original
+dataset. Thus, if you set ``scale=10.0`` and sample without replacement,
+the dataset will have each sample repeated ten times. In general, only
+the last repetition of the dataset will vary when sampling without
+repetition.
+
+Shuffling the data can take considerable CPU time and is turned off by
+default, but is possible by setting the keyword argument ``shuffle=True``.
+
+Finally, for repeatability, you may set the ``random_state`` keyword
+argument.
```

Comment:
I would add a big warning that if you run resample on full data before running cross-validation you will end up with over optimistic results as same samples are likely to appear both in train and test sets.